dbt/models/iasworld/schema/iasworld.asmt_all.yml

```diff
@@ -189,10 +189,88 @@ sources:
+                      AND procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
+                      AND deactivat IS NULL
+                      AND valclass IS NULL
+                      AND class NOT IN ('EX', 'RR')
+                  meta:
+                    table_name: asmt
+                    description: >
+                      valasm1 (ASMT LAND) should be >= 0 if class is not EX or RR
           - name: valasm2
             description: '{{ doc("column_valasm2") }}'
+            tests:
+              - accepted_values:
+                  name: iasworld_asmt_all_valasm2_vact_class_has_val
+                  values: [0]
+                  quote: false
+                  additional_select_columns: *select-columns-with-class
+                  config:
+                    where: |
+                      CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                      AND rolltype != 'RR'
+                      AND procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
+                      AND deactivat IS NULL
+                      AND valclass IS NULL
+                      AND class IN (
+                        '100', '200', '239', '240', '241', '300', '400', '500',
+                        '535', '550', '637', '637A', '637B', '650', '651', '651A',
+                        '651B', '700', '700A', '700B', '742', '742A', '742B',
+                        '800', '800A', '800B', '850', '850A', '850B', '900',
+                        '999', 'EX', 'RR'
+                      )
+                  meta:
+                    table_name: asmt
+                    description: >
+                      valasm2 (ASMT BLDG) should be 0 for vacant classes
+              - not_accepted_values:
+                  name: iasworld_asmt_all_valasm2_impr_class_no_val
+                  values: [0]
+                  quote: false
+                  additional_select_columns: *select-columns-with-class
+                  config:
+                    where: |
+                      CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                      AND rolltype != 'RR'
+                      AND procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
+                      AND deactivat IS NULL
+                      AND valclass IS NULL
+                      AND class NOT IN (
+                        '100', '200', '239', '240', '241', '300', '400', '500',
+                        '535', '550', '637', '637A', '637B', '650', '651', '651A',
+                        '651B', '700', '700A', '700B', '742', '742A', '742B',
+                        '800', '800A', '800B', '850', '850A', '850B', '900',
+                        '999', 'EX', 'RR'
+                      )
+                  meta:
+                    table_name: asmt
+                    description: >
+                      valasm2 (ASMT BLDG) should not be 0 for non-vacant classes
```

Comment:
**suggestion (non-blocking):** It might simplify the description and logic a little bit if this test were something like `valasm2 (ASMT BLDG) should be >= 0 for non-vacant classes`

Reply:
@dfsnow Currently the test is just checking that the value isn't 0, should we change the test along with the description so that it's checking for value > 0?

https://github.com/ccao-data/data-architecture/blob/b57d1a1f538988d00c1c4a7c6a2e6e9ccb539920/dbt/models/iasworld/schema/iasworld.asmt_all.yml#L241-L243

Reply:
I think so. Easier to grok than the double negative.

Reply:
Cool, done in aabc1aa52eccdb5e506c5ab29da5f947a88873c7.